File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice: int = float('inf')
        profit: int = 0
        
        for price in prices:
            minprice = min(minprice, price)
            profit = max(profit, price-minprice)
        
        return profit

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int minprice = numeric_limits<int>::max();
        int maxprofit = 0;
        
        for(int i=0; i<prices.size(); i++){
            minprice = min(minprice, prices[i]);
            maxprofit = max(maxprofit, prices[i]-minprice);
        }
        
        return maxprofit;
        
    }
};